src/loaders/common/callback_server_wrapper.py

```diff
@@ -17,78 +17,121 @@ class Conf:
     """
     Configuration class for the workspace downloader and workspace uploader scripts.
     """
+
     def __init__(
-            self,
-            job_dir: str,
-            output_dir: str,
-            worker_function: Callable,
-            kb_base_url: str = "https://ci.kbase.us/services/",
-            token_filepath: str | None = None,
-            workers: int = 5,
-            retrieve_sample: bool = False,
-            ignore_no_sample_error: bool = False,
+        self,
+        job_dir: str,
+        output_dir: str | None = None,
+        kb_base_url: str = "https://ci.kbase.us/services/",
+        token_filepath: str | None = None,
+        au_service_ver: str = "release",
+        workers: int = 5,
+        max_task: int = 20,
+        worker_function: Callable | None = None,
+        retrieve_sample: bool = False,
+        ignore_no_sample_error: bool = False,
+        workspace_downloader: bool = True,
     ) -> None:
         """
         Initialize the configuration class.
 
         Args:
             job_dir (str): The directory for SDK jobs per user.
             output_dir (str): The directory for a specific workspace id under sourcedata/ws.
-            worker_function (Callable): The function that will be called by the workers.
             kb_base_url (str): The base url of the KBase services.
-            token_filepath (str): The file path that stores a KBase token appropriate for the KBase environment. 
+            token_filepath (str): The file path that stores a KBase token appropriate for the KBase environment.
                                 If not supplied, the token must be provided in the environment variable KB_AUTH_TOKEN.
+            au_service_ver (str): The service verison of AssemblyUtilClient
+                                ('dev', 'beta', 'release', or a git commit).
             workers (int): The number of workers to use for multiprocessing.
+            max_task (int): The maxmium subtasks for the callback server.
+            worker_function (Callable): The function that will be called by the workers.
             retrieve_sample (bool): Whether to retrieve sample for each genome object.
             ignore_no_sample_error (bool): Whether to ignore the error when no sample data is found.
+            workspace_downloader (bool): Whether to be used for the workspace downloader script.
         """
-        port = loader_helper.find_free_port()
-        token = loader_helper.get_token(token_filepath)
-        self.retrieve_sample = retrieve_sample
-        self.ignore_no_sample_error = ignore_no_sample_error
         ipv4 = loader_helper.get_ip()
+        port = loader_helper.find_free_port()
+
+        # common instance variables
+        # the url of the workspace to contact
+        self.ws_url = os.path.join(kb_base_url, "ws")
```

Comment:
This seem like it could also be moved into the downloader section, and removed later

Reply:
removed conf.ws_url in the uploader. Decrease # instance variable by 1

Reply:
:+1: